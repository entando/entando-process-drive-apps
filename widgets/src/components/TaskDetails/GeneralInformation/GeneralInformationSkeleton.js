import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import Skeleton from '@material-ui/lab/Skeleton';
import Divider from '@material-ui/core/Divider';

const styles = {
  divider: {
    marginTop: '20px',
    marginBottom: '15px',
  },
  dataRow: {
    display: 'flex',
    marginTop: '10px',
    marginBottom: '5px',
  },
  value: {
    marginLeft: '10px',
  },
};

const DATA_ROW_HEIGHT = 15;

const GeneralInformationSkeleton = ({ classes }) => (
  <>
    <div>
      <Skeleton width="190px" height={25} variant="rect" />
      <Divider className={classes.divider} />
    </div>
    <div>
      <div className={classes.dataRow}>
        <Skeleton width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="150px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="80px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="150px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="80px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="100px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="150px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
      <div className={classes.dataRow}>
        <Skeleton width="80px" height={DATA_ROW_HEIGHT} variant="rect" />
        <Skeleton className={classes.value} width="190px" height={DATA_ROW_HEIGHT} variant="rect" />
      </div>
    </div>
  </>
);

GeneralInformationSkeleton.propTypes = {
  classes: PropTypes.shape({
    divider: PropTypes.string,
    dataRow: PropTypes.string,
    value: PropTypes.string,
  }).isRequired,
};

export default withStyles(styles)(GeneralInformationSkeleton);
