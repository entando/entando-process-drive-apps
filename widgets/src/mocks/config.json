{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
