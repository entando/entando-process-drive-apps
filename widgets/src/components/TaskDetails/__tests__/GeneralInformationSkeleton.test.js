import { render } from '@testing-library/react';
import React from 'react';
import 'mocks/i18nMock';

import GeneralInformationSkeleton from 'components/TaskDetails/GeneralInformation/GeneralInformationSkeleton';

describe('<GeneralInformationSkeleton />', () => {
  it('renders snapshot correctly', async () => {
    const { container } = render(<GeneralInformationSkeleton />);

    expect(container).toMatchSnapshot();
  });
});
