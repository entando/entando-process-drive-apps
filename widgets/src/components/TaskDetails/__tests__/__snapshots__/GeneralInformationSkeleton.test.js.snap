// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralInformationSkeleton /> renders snapshot correctly 1`] = `
<div>
  <div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="width: 190px; height: 25px;"
    />
    <hr
      class="MuiDivider-root GeneralInformationSkeleton-divider-1"
    />
  </div>
  <div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 150px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 80px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 150px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 80px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 100px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 150px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-dataRow-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 80px; height: 15px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-value-3 MuiSkeleton-pulse"
        style="width: 190px; height: 15px;"
      />
    </div>
  </div>
</div>
`;
